File: src/lessons/turtleart/StarEntity.java
package lessons.turtleart;

import plm.universe.turtles.Turtle;

public class StarEntity extends Turtle {

	/* BEGIN TEMPLATE */
	public void run() {
		/* BEGIN SOLUTION */
		addSizeHint(165, 200, 165, 150);
		for (int i = 0; i < BRANCH_COUNT; i++) 
			branch(50);
		
	}
	static final int BRANCH_COUNT = 5;
	public void branch(int size) {
		forward(size);
		right(360 / BRANCH_COUNT);
		forward(size);

		for (int i = 0; i < 2; i++)
			left(360 / BRANCH_COUNT);
		/* END SOLUTION */
	}
	/* END TEMPLATE */
}

File: resources/exercises/turtleart/star/StarEntity.py
# BEGIN SOLUTION
def branch(size):
  forward(size);
  right(360 / BRANCH_COUNT);
  forward(size);
  #
  for i in range(2):
    left(360 / BRANCH_COUNT);

addSizeHint(165, 200, 165, 150)
BRANCH_COUNT = 5;
for i in range(BRANCH_COUNT):
   branch(50);
# END SOLUTION
